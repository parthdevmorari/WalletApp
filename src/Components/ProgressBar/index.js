import component from './progressBar';
export default component;