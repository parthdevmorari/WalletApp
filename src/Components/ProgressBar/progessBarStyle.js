import { StyleSheet, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color'
import { theme } from '../../Constants/Theme'
const styles = StyleSheet.create({

    mainContainer:{
// alignItems:'center'
paddingLeft:wp(6)
	},
		spendContainer:{
		// width: wp(95),
		paddingVertical:wp(2.5),
		flexDirection: "row",
		justifyContent: "space-between"
	},
	row:{
		flexDirection: "row",

	},
	spendTxt:{
		color: color._222222,
		fontSize: wp(3.5),
		marginTop: wp(5),
		fontWeight: "bold",
	},
	spendTxt2:{
		color: color._01D167,
		fontSize: wp(3.5),
		marginTop: wp(5),
		fontWeight: "bold",
	},
	totalAmount:{
		color: color._555555,
		fontSize: 13,
		marginTop: 20,
		fontWeight: "bold",
		paddingRight: wp(6),
		paddingLeft: wp(1.5),
	},
	amountBox :{
		flexDirection : "row",
	},


});
export default styles;