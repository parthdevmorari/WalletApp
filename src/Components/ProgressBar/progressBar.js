import { View, Text } from 'react-native'
import React from 'react'
import styles from './progessBarStyle'
import * as Progress from 'react-native-progress';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color';

const progressBar = (props) => {
    const { title,spendAmount,totalAmountAvaliable ,progress} = props;

  return (
    
      <View style={[styles.mainContainer]}>
      <View style={[styles.spendContainer]}>
                <Text style={[styles.spendTxt]}>{title}</Text>
                <View style = {styles.amountBox}>
                  <View style={styles.row} >
                <Text style={[styles.spendTxt2]}>{spendAmount}</Text>
                <Text style={[styles.totalAmount]}>{totalAmountAvaliable}</Text>
                </View>
                </View>
            </View>
                <Progress.Bar progress={progress} width={wp(85)} height={wp(3.5)} borderWidth={0} color={color._01D167}unfilledColor={color._e9fcf0} borderRadius={wp(2.5)}/>
               </View>
  )
}

export default progressBar