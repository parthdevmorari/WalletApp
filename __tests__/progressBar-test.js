import 'react-native';
import React from 'react';
import ProgressBar from '../src/Components/ProgressBar/progressBar'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('NoRecord Component', () => {
  test('NoRecord Component', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(wrapper.exists()).toBe(true);
  });
  it('NoRecord  toMatchSnapshot', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});