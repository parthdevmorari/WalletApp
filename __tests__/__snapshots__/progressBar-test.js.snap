// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoRecord Component NoRecord  toMatchSnapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingLeft": 45,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 19,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#222222",
            "fontSize": 26.5,
            "fontWeight": "bold",
            "marginTop": 37.5,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#01D167",
                "fontSize": 26.5,
                "fontWeight": "bold",
                "marginTop": 37.5,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#555555",
                "fontSize": 13,
                "fontWeight": "bold",
                "marginTop": 20,
                "paddingLeft": 11.5,
                "paddingRight": 45,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={19}
    borderWidth={0}
    color="#01D167"
    height={26.5}
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    progress={0}
    unfilledColor="#e9fcf0"
    useNativeDriver={false}
    width={637.5}
  />
</View>
`;
